--- a/net/minecraft/world/level/levelgen/structure/structures/MineshaftPieces.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/MineshaftPieces.java
@@ -42,6 +42,10 @@
 import net.minecraft.world.level.storage.loot.LootTables;
 import org.slf4j.Logger;
 
+// CraftBukkit start - imports
+import net.minecraft.nbt.NBTBase;
+// CraftBukkit end
+
 public class MineshaftPieces {
 
     static final Logger LOGGER = LogUtils.getLogger();
@@ -514,6 +518,8 @@
 
                         if (structureboundingbox.isInside(blockposition_mutableblockposition) && this.isInterior(generatoraccessseed, 1, 0, l, structureboundingbox)) {
                             this.hasPlacedSpider = true;
+                            // CraftBukkit start
+                            /*
                             generatoraccessseed.setBlock(blockposition_mutableblockposition, Blocks.SPAWNER.defaultBlockState(), 2);
                             TileEntity tileentity = generatoraccessseed.getBlockEntity(blockposition_mutableblockposition);
 
@@ -522,6 +528,9 @@
 
                                 tileentitymobspawner.setEntityId(EntityTypes.CAVE_SPIDER, randomsource);
                             }
+                            */
+                            placeCraftSpawner(generatoraccessseed, blockposition_mutableblockposition, org.bukkit.entity.EntityType.CAVE_SPIDER, 2);
+                            // CraftBukkit end
                         }
                     }
                 }